doc/dataModel.md

```diff
@@ -0,0 +1,242 @@
+
+## Final Data Model
+
+This is the final data model that was used in the production when the GPII project ended. It added 4 new fields, the
+red highlighted, in the `Client Credentials` document type to accommodate the requirement for adding NOVA college as
+a new client.
+
+![Initial data model](images/dataModelFinal.png)
+
+## Future Data Model
+
+This data model was in development when the GPII project ended. The goal was to support the user management feature
+implemented using [fluid-express-user](https://github.com/fluid-project/fluid-express-user).
+
+![Data model supporting user management](images/dataModelWithUserManagement.png)
+
+## Table Descriptions
+
+### GPII Keys
+
+| Name | Type | Required or Optional | Description | Default |
+| --- | --- | --- | --- | --- |
+| id | String | Required | The value of the key written onto key tokens. | None |
+| type | String | Required | The document type for storing GPII keys. | The value must be "gpiiKey". |
+| schemaVersion | String | Required | The version of the schema that the current document structure corresponds with. | None |
+| prefsSafeId | String | Optional | The ID of the preferences safe that the GPII key connects with. Set to null if the GPII key does not have a preferences safe connected. | null |
+| prefsSetId | String | Optional | The key of the preferences set in a preferences safe that the GPII key points to. The identifier matches a key within the `contexts` block of a preference set. For example, the default prefsSetId is a fixed name "gpii-default". | null |
+| revoked | Boolean | Required | Whether this GPII key has been revoked. | false |
+| revokedReason | String | Optional | The revoked reason. | null |
+| timestampCreated | Date | Required | The timestamp when the token is created. | now() |
+| timestampUpdated | Date | Optional | The timestamp when the token is updated. | null |
+| timestampRevoked | Date | Optional | The timestamp when the token is revoked. | null |
+
+### Prefs Safes
+
+| Name | Type | Required or Optional | Description | Default |
+| --- | --- | --- | --- | --- |
+| id | String | Required | The ID of the preferences safe. | None |
+| type | String | Required | The document type for storing preferences safes. | The value must be "prefsSafe". |
+| schemaVersion | String | Required | The version of the schema that the current document structure corresponds with. | None |
+| prefsSafeType | enum of ["snapset", "user"] | Required | Indicates whether this preferences safe is allowed to be modified: the ones with "snapset" type are unmodifiable while the ones with "user" type are modifiable. | "user" |
+| name | String | Optional | The user defined name for this preferences safe. A “preference-safe-name and password” pair can potentially be used as a key. | null |
+| password | String | Optional | The user defined password for this preferences safe. A “preference-safe-name and password” pair can potentially be used as a key. | null |
```

Comment:
The `password` field is moved to the `Users` document in the future data model. `password_schema` might be a outdated field that has been removed from `fluid-express-user` repository.

I will update both the field description and the diagram to reflect it.

Reply:
In the latest `dataModelWithUserManagement.png`, neither the `Prefs Safes` model nor the `Users` model have a`password` field.  Since the topic is the *future* data model, I think it's safe to put `password` in the `Users` part of the diagram, and describe it in the `Users` description table.  Also, include a note about it moving there, similar to the note you put in the `Prefs Safes` table where you say that the `password` field is moving to the `Users` document.

Reply:
I believe `Users.derived_key` is the encrypted password. 

@the-t-in-rtf, please correct me if I'm wrong.

Reply:
That makes sense, but I'll wait for confirmation from @the-t-in-rtf

Reply:
That's correct, it's the result of running the original password supplied at signup through the algorithm.  We store that and then if the password supplied at login results in the same value when run through the same algorithm, the passwords are considered to match.